docs/source/geopm_agent_frequency_balancer.7.rst

```diff
@@ -0,0 +1,136 @@
+geopm_agent_frequency_balancer(7) -- Agent reduces imbalance across CPU cores
+=============================================================================
+.. meta::
+
+    :description: The GEOPM frequency_balancer agent reduces imbalance across
+                  CPU cores through frequency controls and performance-guided
+                  turbo limits.
+    :keywords: frequency balancer agent imbalance SST-TF P-States
+
+Description
+-----------
+The ``frequency_balancer`` agent reduces imbalance across CPU cores.
+The executing application's main compute loop must be annotated with
+``geopm_prof_epoch()`` from :doc:`geopm_prof(3) <geopm_prof.3>`. The agent
+assumes that core application time can be measured as time spent outside of
+networking regions of code. CPU cores that spend less networking time per
+epoch are allocated higher CPU frequency limits, while cores that spend more
+networking time per epoch are allocated lower CPU frequency limits.
+
+Allocating CPU Core Frequency Limits
+------------------------------------
+The agent estimates a running application's critical path in each epoch,
+estimates the best-case critical path time the agent can make the application
+achieve (sometimes with energy savings, sometimes with performance improvement),
+and applies frequency control settings that the agent estimates will let each
+core achieve the target best-case time.
+
+The following subsections outline the agent's responsibilities at a high level.
+See `Guiding Hardware-Driven Turbo with Application Performance Awareness <https://ieeexplore.ieee.org/abstract/document/9969356>`_
```

Comment:
Could be useful to put a link to this paper in a docstring in gen_plot_time.py to link the paper to the plotting code.